buildpacks/nodejs-yarn/src/cfg.rs

```diff
@@ -0,0 +1,110 @@
+use heroku_nodejs_utils::{package_json::PackageJson, vrs::Requirement};
+use std::path::Path;
+
+// Reads parsed `engines.yarn` requirement from a `PackageJson`.
+pub(crate) fn requested_yarn_range(pkg_json: &PackageJson) -> Option<Requirement> {
+    pkg_json
+        .engines
+        .as_ref()
+        .and_then(|engines| engines.yarn.clone())
+}
+
+// A yarn cache is populated if it exists, and has non-hidden files.
+pub(crate) fn cache_populated(cache_path: &Path) -> bool {
+    cache_path
+        .read_dir()
+        .map(|mut contents| {
+            contents.any(|entry| {
+                entry
+                    .map(|e| !e.file_name().to_string_lossy().starts_with('.'))
+                    .unwrap_or(false)
+            })
+        })
+        .unwrap_or(false)
+}
+
+// Fetches the build scripts from a `PackageJson` and returns them in priority order
```

Comment:
Nit: these look and feel like docstrings, but are regular comments. Is this by accident or on purpose?

Reply:
Accident. I'll convert them to docstrings.